### PR TITLE
Combined all logins into one login page [WIP]

### PR DESCRIPTION
**Pull request by nlwach**
This PR combines the two different login pages into one. 

It is now possible to login from one page as a student or teacher or directly go to the register page. 

Because of some difficulties with parsing form errors to the single "big" page, after some discussion wit @jnsdrtlf we decided to redirect to the old login screen after a failed login attempt. Here we also show the error in the form. We thought that if one enters a wrong token or key, one most likely wants to try again and one is only interested in the login for which there was a failed attempt. There is also a button to go back to the "big" login page.

The overall functionality is there, the page only needs some styling. Additionally I am not sure what we want to put as text above or below the buttons, so I just copied the text from the old login page and modified it such that it fits the current state of the login page.

### CODE REVIEW

File: gcampus/auth/views/token.py [hunk 1]
```diff
@@ -137,3 +125,30 @@ def logout(request: HttpRequest):
...
+    template_name: str = "gcampusauth/failed_login_form.html"
+    form_class = AccessKeyForm
+    success_url = reverse_lazy("gcampuscore:mapview")
+    token_type = TokenType.access_key
+
+
+class CourseTokenFormView(LoginFormView):
+    template_name: str = "gcampusauth/failed_login_form.html"
+    form_class = CourseTokenForm
+    success_url = reverse_lazy("gcampuscore:mapview")
```
Comment (ax-sc) [inline gcampus/auth/views/token.py:153]: Maybe we could change the success_url to the course overview? I think that´s the page most people want to use after logging in with a course token?

Comment (nlwach) [reply gcampus/auth/views/token.py:153]: Thats a good idea

**Comment by nlwach** [comment]
I noticed that for the "unauthenticated" error pages we're still using the pages where there are three separate buttons for each login and the registration. I changed these now for view/edit permission and unauthenticated error to only display a single login button. For the others it makes sense to only show the login to the specific error pages that we are also redirecting to when a login attempt was unsuccessful (as described above).

**Comment by jnsdrtlf** [comment]
As discussed with @ax-sc, I made some modifications to the login page and put it in its own layout, removing the sidebar, title bar and adding a custom footer.

I am not complelty satisfied with how my layout turned out, but maybe we can improve upon it.

Also a notable change: The form view for course token and access key login now all use the same template.

Thanks @nlwach for the implementation of the login views! Your approach works really well and is clean :)

File: gcampus/auth/templates/gcampusauth/login_form.html [hunk 2]
```diff
@@ -1,62 +1,142 @@
...
+                        to administer the course, edit measurements and
+                        create new access keys.
+                    {% endblocktranslate %}
                 {% endif %}
             </div>
-        </form>
+            <div class="pb-4 row g-0 justify-content-center">
+                <div class="col-12 col-lg-6 px-lg-3 d-flex justify-content-center align-items-start border-end-lg">
+                    <h2>
+                        {% include "gcampuscore/icons/person-fill.html" with extra_class="text-muted flex-shrink-0 me-3" only %}
```
Comment (ax-sc) [inline gcampus/auth/templates/gcampusauth/login_form.html:33]: Can you try to move to the icon to the right, make it bigger and lighter grey? (So we could be more consistent (reusing the style of the water quality index cards.)

File: gcampus/auth/templates/gcampusauth/login_form.html [hunk 3]
```diff
@@ -1,62 +1,142 @@
...
+                            {{ access_key_form.next_url }}
+                            <button class="btn btn-primary btn-block btn-sm w-100" type="submit">
+                                {% translate "Login with access key" %}
+                            </button>
+                        </form>
+                    </div>
+                </div>
+                <div class="col-12 col-lg-6 mt-4 mt-lg-0 px-lg-3 d-flex justify-content-center align-items-start">
+                    <h2>
+                        {% include "gcampuscore/icons/teacher.html" with extra_class="text-muted flex-shrink-0 me-3" only %}
```
Comment (ax-sc) [inline gcampus/auth/templates/gcampusauth/login_form.html:63]: same as the other icon, see comment above.

**Review by ax-sc — CHANGES_REQUESTED** [review]
Thanks to both of you, @jnsdrtlf and @nlwach! I made some comments to the code, let me know if one of them needs more explanation.

I did not spot any problems regarding the functionality, but I am not entirely convinced by the new design altogether yet. I like the left part of it and feel this to be way cleaner and easier to understand now the forms are next to each other. Maybe we can discuss some ideas on how to improve the general layout in our next meeting?

What do you think of removing the sidebar and title bar for the registration form as well? I think this would minimise confusion by having everything "login related" using the same style.

Can you add translations before merging?

**Review by ax-sc — APPROVED** [review]

**Pull request merged**